### PR TITLE
Add Verifalia verify-email action

### PR DESCRIPTION
**Pull request by ecobisi**
This PR adds a new action to the Verifalia app.
Thank you

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3214"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

File: components/verifalia/actions/verify-email/verify-email.mjs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import verifalia from "../../verifalia.app.mjs";
+
+import {
+  CancellationToken,
+  OperationCanceledError,
+} from "verifalia";
+
+export default {
+  name: "Verify Email Address",
+  description: "Verify an email address and check if it is properly formatted, really exists and can accept mails, " +
```
Comment (vellames-turing) [inline components/verifalia/actions/verify-email/verify-email.mjs:11]: Could you please add link to the docs?

File: components/verifalia/actions/verify-email/verify-email.mjs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        }
+
+        // TODO: If the validation completed, abort the suspension and return the results
+        // Blocked by https://github.com/PipedreamHQ/pipedream/issues/3094
+
+        // We are *not* in test mode and the job is complete: simulate an instantaneous
+        // webhook completion callback, as the external one may take a few seconds to
+        // fire and we wish to continue the flow as fast as possible.
+
+        return await axios({
```
Comment (vellames-turing) [inline components/verifalia/actions/verify-email/verify-email.mjs:152]: Could you please move this request to `app` and call it here?

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Hi there, just a couple comments

File: components/verifalia/actions/verify-email/verify-email.mjs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      description: "Enter the email address to verify (e.g. `batman@gmail.com`)",
+      optional: false,
+    },
+    quality: {
+      type: "string",
+      label: "Quality Level",
+      description: "The higher the quality level the longer it could take to complete the verification; by default, " +
+                "we use the configured default quality level for your Verifalia user.",
+      optional: true,
+      options: [
```
[the remark this answers is not in the document]
Comment (ecobisi) [reply components/verifalia/actions/verify-email/verify-email.mjs]: Yes, done and pushed.

File: components/verifalia/actions/verify-email/verify-email.mjs (hunk 2 above)
Comment (ecobisi) [reply components/verifalia/actions/verify-email/verify-email.mjs:141]: I believe this is somewhat overkill and I would prefer to leave it as is; also, that axios request is a temporary hack until https://github.com/PipedreamHQ/pipedream/issues/3151 will be fixed / closed. @vellames-turing could you proceed with the merge anyway, please?

**Review by vellames-turing — APPROVED** [review]

**Comment by vellames-turing** [comment]
@js07 Its failing in some workflows, this is related to the `pnpm` right? do you mind to type the commands to fix the branch, please?

**Comment by js07** [comment]
@vellames-turing  There's a [`post-merge`](https://github.com/PipedreamHQ/pipedream/blob/master/.husky/post-merge) hook in the repo that should run `npx pnpm install -r` and generate a new `pnpm-lock.yaml` if there are any changes to dependencies. Committing and pushing changes to `pnpm-lock.yaml` should fix the failed checks.

From the root of the repo:
```sh
git fetch origin
git merge origin/master
git add pnpm-lock.yaml
git commit
git push
```

**Comment by ecobisi** [comment]
@vellames-turing Sure, just done. Thanks

**Review by vellames-turing — APPROVED** [review]

**Pull request merged**